### PR TITLE
front: skip non-existent paths in gcc depfiles

### PR DESCRIPTION
**Pull request by CyberTailor**
## Summary
There could be non-existent files in `conf.m.fileInfos` that prevented GCC-style depfiles from behaving correctly. This commit filters them out.

## Details
Also create a helper function to quote paths for Unix Makefiles.

### CODE REVIEW

File: compiler/front/depfiles.nim [hunk 1]
```diff
@@ -25,13 +25,14 @@ proc writeDepsFile*(g: ModuleGraph) =
       f.writeLine(toFullPath(g.config, k))
   f.close()
 
+func makefileQuoted(path: string): string =
```
Comment (saem) [inline compiler/front/depfiles.nim]: Isn't there already a pre-existing procedure for this like quote shell?

```suggestion
func quoteFilepath(path: string): string =
```

Also, this seems like a better name

File: compiler/front/depfiles.nim [hunk 2]
```diff
@@ -25,13 +25,14 @@ proc writeDepsFile*(g: ModuleGraph) =
...
   ## Outputs one `make` rule containing target's file name, a colon, and the
   ## names of all specified paths. Spaces and hashes are escaped.
-  let target = target.multiReplace((" ", "\\ "), ("#", "\\#"))
-  depfile.write(target & ": \\" & '\n')
-  for it in paths:
-    let path = it.multiReplace((" ", "\\ "), ("#", "\\#"))
-    if path.len == 0:
+  depfile.write(target.makefileQuoted & ": \\" & '\n')
+  for path in paths:
+    if path.len == 0 or not fileExists(path):
```
Comment (saem) [inline compiler/front/depfiles.nim:37]: I don't recall when this is called, but if this is meant to produce a reusable description then this shouldn't be filtered here.

**Review by saem — CHANGES_REQUESTED** [review]
Thanks for opening a PR.

As per the other PR the template needs to be updated. Some of the key questions as part of the template would surface some deeper insights and help produce a deeper fix.

File: compiler/front/depfiles.nim (hunk 1 above)
Comment (CyberTailor) [reply compiler/front/depfiles.nim]: > Isn't there already a pre-existing procedure for this like quote shell?

Yes, but only subset of that is needed

Comment (CyberTailor) [reply compiler/front/depfiles.nim]: > Also, this seems like a better name

Committed.

File: compiler/front/depfiles.nim (hunk 2 above)
Comment (CyberTailor) [reply compiler/front/depfiles.nim:37]: Updated PR message

**Comment by CyberTailor** [comment]
Closing until I remember what package caused this behavior

**Pull request closed without merging**